x-pack/legacy/plugins/reporting/server/browsers/chromium/driver_factory/index.ts

```diff
@@ -197,7 +230,7 @@ export class HeadlessChromiumDriverFactory {
       });
 
       // unsubscribe logic makes a best-effort attempt to delete the user data directory used by chromium
-      return () => {
```

Comment:
Source of the fix is here, however this is due for further refactoring. This was beyond the scope of fixing the issue in my opinion, but happy to chat more about it.

Reply:
Aha, so the code was there to do it, but it wasn't working properly